File: example/1d-linear-convection/weno3/python/04i/mesh.py
# mesh.py
import numpy as np

# Mesh class: defines computational grid
class Mesh:
    def __init__(self):
        self.xmin = 0.0
        self.xmax = 2.0
        self.ncells = 40
        self.nnodes = self.ncells + 1
        self.nx = self.ncells
        self.x = np.zeros(self.nnodes)
        self.xcc = np.zeros(self.ncells)
        self.init_mesh()
        
    def init_mesh(self):
        self.L = self.xmax - self.xmin
        self.dx = self.L / self.ncells
        
        # Generate node coordinates
        for i in range(self.nnodes):
            self.x[i] = self.xmin + i * self.dx
            
        # Generate cell center coordinates
        for i in range(self.ncells):
            self.xcc[i] = 0.5 * (self.x[i] + self.x[i+1])

File: example/1d-linear-convection/weno3/julia/02e/mesh.jl
# julia/mesh.jl
"""
Mesh 结构体（与提供的 mesh.py 完全同构）
- 字段名、初始化顺序、循环逻辑完全一致
- 不做任何泛化或优化
"""

mutable struct Mesh
    xmin::Float64
    xmax::Float64
    ncells::Int
    nnodes::Int
    nx::Int
    x::Vector{Float64}
    xcc::Vector{Float64}
    L::Float64      # 在 init_mesh 中赋值
    dx::Float64     # 在 init_mesh 中赋值
    
    function Mesh()
        # 与 Python __init__ 完全一致
        xmin = 0.0
        xmax = 2.0
        ncells = 40
        nnodes = ncells + 1
        nx = ncells
        x = zeros(Float64, nnodes)
        xcc = zeros(Float64, ncells)
        
        # 调用 init_mesh（模拟 Python）
        L = xmax - xmin
        dx = L / ncells
        
        # Generate node coordinates: for i in range(self.nnodes)
        for i in 1:nnodes
            x[i] = xmin + (i - 1) * dx   # Python i → Julia i-1
        end
        
        # Generate cell center coordinates
        for i in 1:ncells
            xcc[i] = 0.5 * (x[i] + x[i+1])
        end
        
        new(xmin, xmax, ncells, nnodes, nx, x, xcc, L, dx)
    end
end